src/strands/types/tools.py

```diff
@@ -91,7 +91,7 @@ class ToolResult(TypedDict):
     """
 
     content: list[ToolResultContent]
-    status: ToolResultStatus
+    status: NotRequired[Optional[ToolResultStatus]]
```

Comment:
The NotRequired[Optional] Seems odd here; should we just be marking it as `NotRequired`?